include/mrdox/Metadata/TemplateParam.hpp

```diff
@@ -21,6 +23,22 @@
 namespace clang {
 namespace mrdox {
 
+
+enum class TemplateParamKind
+{
+    // KRYSTIAN NOTE: temporary
+    None,
+    // template type parameter, e.g. "typename T" or "class T"
+    Type,
+    // template non-type parameter, e.g. "int N" or "auto N"
+    NonType,
+    // template template parameter, e.g. "template<typename> typename T"
+    Template,
+};
```

Comment:
Concepts are `Type` ?

Reply:
We don't handle concepts at all currently. Once I add partial/explicit specialization support, I'll add support for concepts & requirement clauses

Reply:
concepts are a whole other ball of yarn